src/main/java/net/citizensnpcs/npc/entity/HumanController.java

```diff
@@ -1,57 +1,62 @@
 package net.citizensnpcs.npc.entity;
 
-import java.lang.reflect.Method;
-import java.net.URL;
-import java.util.Iterator;
-import java.util.Map;
 import java.util.UUID;
-import java.util.concurrent.BlockingDeque;
-import java.util.concurrent.Callable;
-import java.util.concurrent.LinkedBlockingDeque;
 import java.util.regex.Pattern;
 
-import org.bukkit.Bukkit;
-import org.bukkit.ChatColor;
-import org.bukkit.Location;
-import org.bukkit.craftbukkit.v1_8_R3.CraftServer;
-import org.bukkit.craftbukkit.v1_8_R3.CraftWorld;
-import org.bukkit.entity.Entity;
-import org.bukkit.entity.Player;
-import org.bukkit.scoreboard.Scoreboard;
-import org.bukkit.scoreboard.Team;
-
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Maps;
-import com.mojang.authlib.Agent;
 import com.mojang.authlib.GameProfile;
-import com.mojang.authlib.GameProfileRepository;
-import com.mojang.authlib.HttpAuthenticationService;
-import com.mojang.authlib.ProfileLookupCallback;
-import com.mojang.authlib.minecraft.MinecraftSessionService;
-import com.mojang.authlib.properties.Property;
-import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
-import com.mojang.authlib.yggdrasil.YggdrasilMinecraftSessionService;
-import com.mojang.authlib.yggdrasil.response.MinecraftProfilePropertiesResponse;
-import com.mojang.util.UUIDTypeAdapter;
-
-import net.citizensnpcs.Settings.Setting;
+
 import net.citizensnpcs.api.CitizensAPI;
-import net.citizensnpcs.api.event.DespawnReason;
 import net.citizensnpcs.api.npc.NPC;
 import net.citizensnpcs.api.util.Colorizer;
-import net.citizensnpcs.api.util.Messaging;
 import net.citizensnpcs.npc.AbstractEntityController;
+import net.citizensnpcs.npc.skin.Skin;
+import net.citizensnpcs.npc.skin.EntitySkinnable;
 import net.citizensnpcs.util.NMS;
 import net.minecraft.server.v1_8_R3.PlayerInteractManager;
 import net.minecraft.server.v1_8_R3.WorldServer;
 
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.craftbukkit.v1_8_R3.CraftWorld;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.scoreboard.Scoreboard;
+import org.bukkit.scoreboard.Team;
+
 public class HumanController extends AbstractEntityController {
+
     public HumanController() {
         super();
-        if (SKIN_THREAD == null) {
-            Bukkit.getScheduler().runTaskTimerAsynchronously(CitizensAPI.getPlugin(), SKIN_THREAD = new SkinThread(),
-                    10, 10);
+    }
+
+    @Override
+    protected boolean couldSpawn(Location at, NPC npc) {
+        if (!super.couldSpawn(at, npc)) {
+            return false;
         }
+
+        EntitySkinnable skinnable = NMS.getSkinnableNPC(npc.getEntity());
+        assert skinnable != null;
+
+        skinnable.getSkinTracker().updateNearbyViewers(100.0);
```

Comment:
100 constant seems arbitrary but used throughout, maybe need to merge into somewhere (possibly a setting).